modules/docs/lib/widgets/enhancedComponent.tsx

```diff
@@ -0,0 +1,200 @@
+import {MdxJSToJSX, MDX} from '../MDXElements';
+import {SymbolDoc} from '../SymbolDoc';
+import {PropertiesTable, registerWidget, Value} from '../Value';
+import {Heading, SymbolDialog} from '../widgetUtils';
+import * as types from '../../docgen/docTypes';
+
+const ParentComponentNameContext = React.createContext('');
+/**
+ * Used to keep track of JSDoc in subcomponent definitions.
+ *
+ * For example:
+ * ```tsx
+ * subComponents: {
+ *   // JSDoc goes here instead of on `TabsItem` for better dev experience
+ *   Item: TabsItem
+ * }
+ * ```
+ */
+const ParentComponentJSDocContext = React.createContext(defaultJSDoc);
+
+const fileNameToCategoryMap: Record<string, string[]> = {
+  Layout: [
+    'utils/layout',
+    'utils/gridArea',
+    'utils/gridItem',
+    'utils/flexItem',
+    'utils/position',
+    'utils/space',
+    'utils/stack',
+  ],
+  Color: ['utils/color', 'utils/background'],
+  Border: ['utils/border'],
+  Depth: ['utils/depth'],
+  Text: ['utils/text'],
+  Other: ['utils/other'],
+};
+
+registerWidget<EnhancedComponentValue>('enhancedComponent', ({value, doc, meta}) => {
+  const groups = groupProps(value.props);
+  const parentComponentName = React.useContext(ParentComponentNameContext);
+  const parentComponentJSDoc = React.useContext(ParentComponentJSDocContext);
+  const hideHeader = meta && meta.hideHeader === true;
+
+  const intro = (
+    <>
+      {!parentComponentName && !hideHeader && value.displayName ? (
+        <Heading>{value.displayName}</Heading>
+      ) : null}
+
+      <ParentComponentNameContext.Provider value="">
+        <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+          <MdxJSToJSX>{parentComponentJSDoc.description || doc?.description || ''}</MdxJSToJSX>
+        </ParentComponentJSDocContext.Provider>
+      </ParentComponentNameContext.Provider>
+    </>
+  );
+
+  // We don't want to re-document a subcomponent that is a container component with a different
+  // model. If we detect another container, only render the intro and a link to the container
+  // component
+  if (parentComponentName && value.componentType === 'container') {
+    return (
+      <>
+        {intro}
+        <MDX as="p">
+          This component references the{' '}
+          <ParentComponentNameContext.Provider value="">
+            <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+              <SymbolDialog value={{kind: 'symbol', name: value.displayName || ''}} />
+            </ParentComponentJSDocContext.Provider>
+          </ParentComponentNameContext.Provider>{' '}
+          component.
+        </MDX>
+      </>
+    );
+  }
+
+  return (
+    <>
+      {intro}
+      {value.styleComponent ? (
+        <>
+          <Heading headingOffset={1}>Style Component</Heading>
+          <MDX as="p">
+            <code>
+              <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+                <Value value={value.styleComponent} />
```

Comment:
I'm seeing an issue with the StyleComponent popup where it's showing duplicate content:

<img width="1251" alt="Screenshot 2023-02-28 at 6 37 25 PM" src="https://user-images.githubusercontent.com/146020/222031065-9a8ba9f3-98e1-4a19-9697-afcb0e8008e1.png">

Reply:
I can fix this specific instance, but I still have a problem with `@link`s that reference components.

The `SymbolDoc` renders JSDoc descriptions, but a widget might also render JSDoc descriptions. I need a way to track when this is being done...

Reply:
The problem I run into is sometimes I need the `name` to render before the `description` and sometimes I don't want the name to render.

Reply:
For example, in a `SymbolDialog`, the name is rendered in the header and should not be rendered in the `Value`

Also, not all `Value`s have a JSDoc or a `name` property...

There should be better heuristics though. It feels like I'm chasing edge cases constantly